packages/devtools_app/lib/src/provider/provider_screen.dart

```diff
@@ -0,0 +1,805 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:math' as math;
+
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:pedantic/pedantic.dart';
+
+import '../common_widgets.dart';
+import '../config_specific/logger/logger.dart';
+import '../eval_on_dart_library.dart';
+import '../globals.dart';
+import '../inspector/inspector_text_styles.dart';
+import '../inspector/inspector_tree.dart';
+import '../screen.dart';
+import '../split.dart';
+import 'eval.dart';
+import 'provider_state_controller.dart';
+
+const kTypeColor = Color.fromARGB(255, 78, 201, 176);
+const kBoolColor = Color.fromARGB(255, 86, 156, 214);
+const kNullColor = kBoolColor;
+const kNumColor = Color.fromARGB(255, 181, 206, 168);
+const kStringColor = Color.fromARGB(255, 206, 145, 120);
+const kPropertyColor = Color.fromARGB(255, 206, 145, 120);
+
+const double rowHeight = 20.0;
+const double horizontalSpacing = 15;
+const double expandIconSize = 20;
+
+final _providerIdProvider = ScopedProvider<String>(null);
+
+final AutoDisposeStateProvider<String> _selectedProviderIdProvider =
+    AutoDisposeStateProvider<String>((ref) {
+  final providerIdsStream = ref.watch(providerIdsProvider.stream);
+
+  StreamSubscription sub;
+  sub = providerIdsStream.where((ids) => ids.isNotEmpty).listen((ids) {
+    sub.cancel();
+    ref.read(_selectedProviderIdProvider).state = ids.first;
+  });
+
+  ref.onDispose(sub.cancel);
+
+  return null;
+});
+
+final _hasSelectedProviderProvider = AutoDisposeProvider<bool>((ref) {
+  return ref.watch(_selectedProviderIdProvider).state != null;
+});
+
+final _isSelectedProvider = ScopedProvider<bool>((watch) {
+  return watch(_selectedProviderIdProvider).state == watch(_providerIdProvider);
+});
+
+final _isExpandedProvider =
+    AutoDisposeStateProviderFamily<bool, InstancePath>((ref, path) {
+  // TODO refreshing the provider (after evaluating an expression) should not reset the expansion state
+  // expands the root by default, but not children
+  return path.pathToProperty.isEmpty;
+});
+
+final _selectedProviderEvalProvider =
+    AutoDisposeFutureProvider<EvalOnDartLibrary>((ref) async {
+  final isAlive = IsAlive();
+  ref.onDispose(isAlive.dispose);
+
+  final selectedProviderId = ref.watch(_selectedProviderIdProvider).state;
+
+  final instanceDetails = await ref.watch(
+    instanceProvider(InstancePath.fromProvider(selectedProviderId)).future,
+  );
+
+  return instanceDetails.maybeMap(
+    object: (instance) => instance.evalForInstance,
+    orElse: () => ref.watch(evalProvider),
+  );
+});
+
+class ProviderScreen extends Screen {
+  const ProviderScreen()
+      : super.conditional(
+          id: id,
+          requiresLibrary: 'package:provider/',
```

Comment:
nit: if there is a specific library only available under package:provider when you provide the support this screen needs you could make it conditional on that library.

Reply:
Only provider is required, but it needs a specific version constraints (probably >=5.0.0-nullsafety.x)